modules/nf-core/bwa/mem/main.nf

```diff
@@ -24,6 +28,13 @@ process BWA_MEM {
     def args2 = task.ext.args2 ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
     def samtools_command = sort_bam ? 'sort' : 'view'
+    def extension = args2.contains("--output-fmt sam")   ? "sam" :
+                    args2.contains("--output-fmt cram")  ? "cram":
+                    sort_bam && args2.contains("-O cram")? "cram":
+                    !sort_bam && args2.contains("-C")    ? "cram":
+                    "bam"
```

Comment:
What about `-O bam` and I think the extension can also be uppercase, right? (Sorry for all the nitpicking ehehe :grin:)

Reply:
Anything not cram defaults to bam, so no need to check

Reply:
Isn't the default SAM?

Reply:
samtools infers the bam format from the `-o ${prefix}.bam`